src/ploneintranet/workspace/browser/tiles/sidebar.py

```diff
@@ -70,12 +70,80 @@ def can_manage_workspace(self):
     def can_add(self):
         """
         Is this user allowed to add content?
+        Cave. This was an easy way when we had archetypes.
+        With Dexterity, each content type is protected by its own
+        permission. Use more specific checks, if you can.
         """
         return api.user.has_permission(
             "Add portal content",
             obj=self.context,
         )
 
+    def can_add_documents(self):
```

Comment:
Can we add a @memoize decorator from plone.memoize.view?
Those methods are called many times in the template.